### PR TITLE
Disallow importing handlers code from backends

### PR DESCRIPTION
**Pull request by AlekSi**
# Description

Except `sjson`.

## Readiness checklist

- [ ] I added/updated unit tests (and they pass).
- [ ] I added/updated integration/compatibility tests (and they pass).
- [ ] I added/updated comments and checked rendering.
- [ ] I made spot refactorings.
- [ ] I updated user documentation.
- [x] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

**Review by chilagrow — APPROVED** [review]
thank you!

File: .golangci.yml [hunk 1]
```diff
@@ -46,6 +46,11 @@ linters-settings:
           - "!**/internal/handlers/hana/hanadb/*.go"
         deny:
           - pkg: github.com/FerretDB/FerretDB/internal/handlers/sjson
+      backends:
+        files:
+          - "**/internal/backends/**/*.go"
```
Comment (rumyantseva) [inline .golangci.yml:51]: Is ` **/internal/backends/*.go` also included?

**Review by rumyantseva — APPROVED** [review]
Looks reasonable!

File: .golangci.yml (hunk 1 above)
Comment (AlekSi) [reply .golangci.yml:51]: `**` expands into anything, including an empty string and sub/dir/ectory

**Pull request merged**